src/main/rx/fport.c

```diff
@@ -207,25 +207,20 @@ static void fportDataReceive(uint16_t c, void *data)
                 return;
             }
 
-            if (bufferPosition == 1 && val == FPORT_FRAME_TYPE_TELEMETRY_REQUEST) {
+            if (framePosition == 2 && val == FPORT_FRAME_TYPE_TELEMETRY_REQUEST) {
                 telemetryFrame = true;
             }
 
-            frameStarting = false;
-            rxBuffer[rxBufferWriteIndex].data[bufferPosition++] = val;
-        } else if (bufferPosition >= BUFFER_SIZE) {
-            bufferPosition = 0;
-
-            reportFrameError(DEBUG_FPORT_ERROR_OVERSIZE);
+            rxBuffer[rxBufferWriteIndex].data[framePosition - 1] = val;
+            framePosition = framePosition + 1;
         }
-
     }
 }
 
 #if defined(USE_TELEMETRY_SMARTPORT)
 static void smartPortWriteFrameFport(const smartPortPayload_t *payload)
 {
-    skipUntilStart = true;
+    framePosition = 0;
```

Comment:
Only race condition will cause nonzero `framePosition` here (clearToSend is cleared on frame start). But killing receive is probably best way to handle it.

Reply:
No, absolutely not. Due to the fact that FPort is half duplex, every byte sent will be received as well (at least on F4, F3 seems to be filtering out the feedback bytes).

Reply:
Yes, but reply does not contain FPORT_FRAME_MARKER, and receiver must be in 'interframe' mode (framePosition == 0).
So `framePosition = 0` has any effect only on race condition, but it is good idea to handle race this way.

Reply:
Ah yes, race condition between the RX and the flight controller. That is exactly what I am worried about, since we don't have control over the RX, and don't know how its output will be affected by things like radio link loss and subsequent resynch. I suspect it's not actually keeping to the timing it had before the link was lost in this case, but resynching on the new radio packet stream.

Reply:
The race window is much smaller - it will happen only when character is received after checking `clearToSend`, but before getting here (few instructions).
Line collision will happen anyway (and window for collision is considerably longer - whole byte must be received to clear clearToSend), but `framePosition = 0` may help receiver to discard resulting garbage.